### ISSUE TITLE
Improve Python/NumPy code

### ISSUE THREAD

**Issue by kir0ul** [NiloufarRazmi/RL_Olfaction#1]
At the moment, to convert the `Cell` data structure from Matlab, standard Python lists are used. Could be worth using `Awkward Array` to make computations faster and allow some broadcasting, see:
- https://awkward-array.org/
- https://github.com/jpivarski-talks/2020-02-27-irishep-poster/blob/master/pivarski-irishep-poster.pdf
- https://github.com/jpivarski-talks/2022-07-25-cssi-meeting-poster/blob/main/pivarski-awkward-cssi-poster.pdf

**Comment by kir0ul** [comment]
Maybe [`activity[j]`](https://github.com/NiloufarRazmi/RL_Olfaction/blob/main/backprop/backprop.py#L168) and [`delta[j + 1]`](https://github.com/NiloufarRazmi/RL_Olfaction/blob/main/backprop/backprop.py#L211) could be converted to a matrix to remove the need to check for the dimension of the array.

**Comment by kir0ul** [comment]
The code could be abstracted a bit more to higher level concepts, e.g., it could make sense to call higher level functions like a `forward_pass()`, `backprop()`, `activation()` (instead of copy/pasting several times the full length sigmoid function).

Could borrow some ideas from this reference code: https://github.com/mnielsen/neural-networks-and-deep-learning/blob/master/src/network.py, taken from [Neural Networks and Deep Learning by Michael Nielsen](http://neuralnetworksanddeeplearning.com/)
